arraycontext/impl/pytato/__init__.py

```diff
@@ -546,9 +661,12 @@ def _to_frozen(key: tuple[Any, ...], ary) -> TaggableCLArray:
                     self._dag_transform_cache[normalized_expr])
 
         assert len(pt_prg.bound_arguments) == 0
-        _evt, out_dict = pt_prg(self.queue,
+        evt, out_dict = pt_prg(self.queue,
                 allocator=self.allocator,
                 **bound_arguments)
+
+        self.add_profiling_event(evt, pt_prg)
```

Comment:
This is inaccurate: one loopy `TranslationUnit` will enqueue multiple kernels, so you'll need to either
- collect multiple events, or
- enqueue a marker before and after.

(Also below.)

Reply:
I implemented the "marker" strategy in [7216ccf](https://github.com/inducer/arraycontext/pull/311/commits/7216ccf34e16cb390170d5251848f90eb2787406)